### ISSUE TITLE
Remove validation for registered system paths from Menu System

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#3945]
**Description of the bug**
I'm in a little over my head here. But, I'll report this and we'll see if it makes sense. 

As I [described in the forum](https://forum.backdropcms.org/forum/how-create-menu-link-pdf) I tried to create a menu link directly to a PDF in the file system, but I get an error message saying "The path '...' is either invalid or you do not have access to it."

I would expect to be able to create a menu link to a file uploaded to Backdrop. 

![Edit_link___Core](https://user-images.githubusercontent.com/3144571/61611164-bbba7200-ac20-11e9-834a-303bf8ce4836.png)

[In this issue](https://github.com/backdrop-contrib/special_menu_items/issues/1#issuecomment-513625349) @jenlampton said the following: 
> the menu system checks links to make sure they are registered system paths. We need to remove that validation too.

Which lead me to open this ticket and call it a bug (which might not be accurate). 

**Steps To Reproduce**
To reproduce the behavior:
1. Create a file field on a content type.
2. Add a file to a node and save
3. copy the relative path to the file in the file directory
4. Try to create a menu item linking to that file.

**Comment by jenlampton** [comment]
There may be another issue about this already. The original use-case was to allow people to create the entire menu before any of the content, layouts, or views existed. Then add those items in later.

**Comment by klonos** [comment]
> ...open this ticket and call it a bug (which might not be accurate).

Title change makes it a bug 😉 

@jenlampton the issues you are referring to are #434 and #422, but these are different to the problem described here.

**Comment by jenlampton** [comment]
Not really - the system was never designed to do this, so it's still a feature request. And the same code change would solve both problems...

**Comment by klonos** [comment]
> ...the system was never designed to do this, so it's still a feature request.

If by "system" you mean old Drupal versions, and by "not designed to do this" you mean path validation failing to recognize *existing* file paths, then I agree; but that does not make it a feature request. At the very least, that error message saying that the path is invalid, or that the user does not have permissions to access it, is incorrect in this case; hence a UX bug.

The difference with #422 is that in that issue we will try to make it so that validation is ignored for *all paths*, while this issue here is to accept valid/existing paths for what they are, and not complain incorrectly.

Having said that, I understand that implementing what is being discussed over in #422 may make this issue here moot. We just need to account fo valid/existing **file** paths in the warning we throw (if that will be what we go with there).

Lets postpone this issue over #422 then.

**Comment by jenlampton** [comment]
> that error message saying that the path is invalid, or that the user does not have permissions to access it, is incorrect in this case

A URL to a file is not actually a "valid system path" so the error here is technically correct. The problem is that any normal human being would (and should) think that a path to a file *is* valid. 

UX bug - sure. But also a feature addition :)

**Comment by klonos** [comment]
Agreed; I tend to think things from a non-coder PoV, so was all bug to me 🙂
